common-compose.yml

```diff
@@ -0,0 +1,53 @@
+# Elasticsearch stack designed for container-native deployment
+# on Joyent's Triton platform.
+
+# ---------------------------------------------------
+# This common service acts as a template for master-only, data-only, and
+# master+data ES services defined in the docker-compose.yml file.
+
+elasticsearch:
+    image: 0x74696d/triton-elasticsearch:latest
+    mem_limit: 4g
+    restart: always
+
+    environment:
+    - CLUSTER_NAME=elasticsearch
+    - ES_HEAP_SIZE=1g # set to 50% of mem_limit, but no more than 31g
+    - ES_NODE_MASTER=true # default
+    - ES_NODE_DATA=true   # default
+    - CONTAINERBUDDY=file:///etc/containerbuddy.json
+    # ES_HEAP_NEWSIZE # optional override
+    # ES_DIRECT_SIZE  # optional override
+    # ES_JAVA_OPTS    # optional override
```

Comment:
Actually, grabbing those values is the easy part, but injecting them into the environment that we fork for the application will involve some new hairiness in Containerbuddy, if it's even an option (I suspect not). I think we can simply pre-calculate and then add these to the `docker-compose` environment, with a note on how to generate them in the example.